java/com/google/flatbuffers/FlexBuffers.java

```diff
@@ -0,0 +1,830 @@
+         * @return blob as a byte array
+         */
+        public byte[] getBytes() {
+            int size = size();
+            byte[] result = new byte[size];
+            for (int i = 0; i < size; i++) {
+                result[i] = bb.get(end + i);
+            }
+            return result;
+        }
+
+        @Override
+        public String toString() {
+            return Utf8.getDefault().decodeUtf8(bb, end, size());
+        }
+
+        @Override
+        public StringBuilder toString(StringBuilder sb) {
+            sb.append('"');
+            sb.append(Utf8.getDefault().decodeUtf8(bb, end, size()));
+            return sb.append('"');
+        }
+    }
+
+    public static class Key extends Object {
+
+        private static final Key EMPTY = new Key(EMPTY_BB, 0, 0);
+
+        Key(ByteBuffer buff, int end, int byteWidth) {
+            super(buff, end, byteWidth);
+        }
+
+        public static Key empty() {
+            return Key.EMPTY;
+        }
+
+        @Override
+        public StringBuilder toString(StringBuilder sb) {
+            int size;
+            for (int i = end; ; i++) {
+                if (bb.get(i) == 0) {
+                    size = i - end;
+                    break;
+                }
+            }
+            sb.append(Utf8.getDefault().decodeUtf8(bb, end, size));
+            return sb;
+        }
+
+        int compareTo(byte[] other) {
+            int ia = end;
+            int io = 0;
+            byte c1, c2;
+            do {
+                c1 = bb.get(ia);
+                c2 = other[io];
+                if (c1 == '\0')
+                    return c1 - c2;
+                ia++;
+                io++;
+                if (io == other.length) {
+                    // in our buffer we have an additional \0 byte
+                    // but this does not exist in regular Java strings, so we return now
+                    return c1 - c2;
+                }
+            }
+            while (c1 == c2);
+            return c1 - c2;
+        }
+
+        @Override
+        public boolean equals(java.lang.Object obj) {
+            if (!(obj instanceof Key))
+                return false;
+
+            return ((Key) obj).end == end && ((Key) obj).byteWidth == byteWidth;
+        }
+    }
+
+    /**
+     * Map object representing a set of key-value pairs.
+     */
+    public static class Map extends Vector {
+        private static final Map EMPTY_MAP = new Map(EMPTY_BB, 0, 0);
+
+        Map(ByteBuffer bb, int end, int byteWidth) {
+            super(bb, end, byteWidth);
+        }
+
+        public static Map empty() {
+            return EMPTY_MAP;
+        }
+
+        /**
+         * @param key access key to element on map
+         * @return reference to value in map
+         */
+        public Reference get(String key) {
+            KeyVector keys = keys();
+            int size = keys.size();
+            int index = binarySearch(keys, key.getBytes(StandardCharsets.UTF_8));
```

Comment:
This is unfortunate, having to convert a key to UTF8 on each map access. Maybe (or in the future) we can provide an alternative version of this function that allows lookup with existing UTF8 data in a `byte[]`

Reply:
I added `Reference get(String key)` as well